src/hardware/input/mouseif_dos_driver.cpp

```diff
@@ -1280,8 +1278,17 @@ void MOUSEDOS_NotifyMoved(const float x_rel, const float y_rel,
 		// Uses absolute mouse position (seamless mode), relative
 		// movements can wait to be reported - they are completely
 		// unreliable anyway
-		if (pending.x_abs != x_abs || pending.y_abs != y_abs)
+		constexpr float Epsilon = 0.5f;
+		if (std::lround(pending.x_abs / Epsilon) != std::lround(x_abs / Epsilon) ||
+		    std::lround(pending.y_abs / Epsilon) != std::lround(y_abs / Epsilon)) {
 			event_needed = true;
+		}
+		// TODO: Consider introducing some kind of sensitivity to avoid
+		// unnecessary events, for example calculated using
+		// 'state.maxpos_*' and 'mouse_shared.resolution_*' values.
+		// Problem: when the mouse is moved really fast and it leaves
+		// the window, the guest cursor is sometimes left like 30 pixels
+		// from the window's edge; we need a mitigation mechanism here
```

Comment:
I'd say that's fine. Solution: don't move it very fast if you don't want the pointer to stop before the edge? 😄 

Btw, I don't see why it matters; maybe for seamless mode?